### DISCUSSION TITLE
Questions about dometyl

### DISCUSSION THREAD

**Discussion by dohn-joh** [geoffder/dometyl-keyboard#9] [Q&A]
1. Setup is no problem, however I am not a programmer. I have only been exposed to a little python and no ocaml. How difficult would it be to use dometyl for a beginner like me or would it not be recommended to attempt using it?
2. I would like to add some rows and columns (5x6) as well as an additional thumb row. Does configuration support this?
3. Does dometyl have choc hotswap support?

**Comment by geoffder** [comment]
Sorry for the delay, forgot to get back to answering this.

1. I wouldn't recommend against giving it a try, but I would strongly recommend getting your editor set up for ocaml so that the language server can help point out mistakes and give you easy access to the docs for the functions that you'll need to use. A lot of it will just be changing some numbers though so hopefully not too daunting.
2. Yes, adding rows and columns can be done by changing the keyword parameters `~n_body_cols` and `~n_body_rows` on the `Plate.make` function (seen in action in the [skeletyl example](https://github.com/geoffder/dometyl-keyboard/blob/d6af46ea157a38167b50cc492e16428644806ddb/models/examples/skeletyl.ml#L38). Thumb rows are adjusted similarly (`~n_thumb_cols`, `~n_thumb_rows`).
3. Yes, choc hotswap socket insets are supported.

**Reply by dohn-joh** [reply]
Thanks for the reply! I may try this in the upcoming months. I will let you know how it works out or if I run into any issues.
